lpc55_sign/src/debug_auth.rs

```diff
@@ -0,0 +1,213 @@
+use lpc55_areas::{CCSOCUDflt, CCSOCUPin, DebugSettings};
+use num_traits::ToPrimitive;
+use rsa::{PublicKeyParts, RsaPrivateKey, RsaPublicKey};
+use sha2::{Digest, Sha256};
+use x509_cert::Certificate;
+
+use crate::{
+    cert::public_key,
+    signed_image::{pad_roots, root_key_hash},
+    Error,
+};
+
+const SOCC: u32 = 0x0000_0001;
+
+#[derive(Debug)]
+pub struct DebugAuthChallenge {
+    // NXP UM11126 claims this is a single little-endian u32 version field.
+    // NXP's spsdk tooling implements it as done here: two little-endian u16s
+    // with version_major at the lower offset.  See
+    // https://github.com/nxp-mcuxpresso/spsdk/blob/5da31d96a020bd65e5834ea5ac1b68327ea965ef/spsdk/dat/dac_packet.py#L114.
+    version_major: u16,
```

Comment:
Consider `pub`-ing these fields to make tool access easier and suppress the unused warning as a side effect.